packages/redux/src/contents/utils.ts

```diff
@@ -20,6 +21,153 @@ export const ENVIRONMENT_CODES = {
   PREVIEW: 'preview',
 };
 
+/**
+ * Generate a ranking number for each commerce page.
+ *
+ * @function
+ *
+ * @param {object} metadata - Metadata object with custom attributes.
+ * @returns {number} Ranking number for a specific commerce page.
+ *
+ * @example
+ * const pageRanking = getPageRanking({ custom: { type: "listing" } });
+ * Result of pageRanking === 0;
+ */
+export const getPageRanking = (metadata: Metadata): number => {
+  let ranking = 0;
+  const { gender, brand, category, priceType, id } = metadata.custom;
+
+  // Return default ranking number when no data was returned from each commerce property.
+  // We can't only check if it's empty from 'metadata.custom' because it could have other properties inside.
+  if (!gender && !brand && !category && !priceType && !id) {
+    return ranking;
+  }
+
+  const rankingGender = 10;
+  const rankingBrand = 100;
+  const rankingCategory = 10000;
+  const rankingPriceType = 1000000000;
+  const rankingId = 10000000000;
+
+  Object.entries(metadata.custom).forEach(entry => {
+    const [key, value] = entry;
+
+    switch (key) {
+      case 'gender':
+        !isEmpty(value) && (ranking += rankingGender);
```

Comment:
What if you remove this verification from each line and just add a return before the `switch`? Something like

```js
if (isEmpty(value)) {
  return ranking;
}
```

Reply:
Done.